src/thunderbots/software/test/primitive/move_primitive.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * This file contains the unit tests for the MovePrimitive class
+ */
+
+#include "ai/primitive/move_primitive.h"
+
+#include <gtest/gtest.h>
+#include <string.h>
+
+TEST(MovePrimTest, construct_with_no_params_test)
+{
+    const std::string move_prim_name = "Move Primitive";
+
+    MovePrimitive move_prim = MovePrimitive(int(), Point(), Angle(), double());
+
+    EXPECT_EQ(int(), move_prim.getRobotId());
+    EXPECT_EQ(move_prim_name, move_prim.getPrimitiveName());
+}
+
+TEST(MovePrimTest, get_robot_id_test)
+{
+    int robot_id = 4U;
+
+    MovePrimitive move_prim = MovePrimitive(robot_id, Point(), Angle(), double());
+
+    EXPECT_EQ(robot_id, move_prim.getRobotId());
+}
+
+TEST(MovePrimTest, parameter_array_test)
+{
+    const Point destination  = Point(-1, 2);
+    const Angle final_angle  = Angle::ofRadians(3.15);
+    const double final_speed = 2.11;
+    const int robot_id       = 2U;
+
+    MovePrimitive move_prim =
+        MovePrimitive(robot_id, destination, final_angle, final_speed);
+
+    std::vector<double> param_array = move_prim.getParameterArray();
+
+    EXPECT_DOUBLE_EQ(destination.x(), param_array[0]);
+    EXPECT_DOUBLE_EQ(destination.y(), param_array[1]);
+    EXPECT_DOUBLE_EQ(final_angle.toRadians(), param_array[2]);
+    EXPECT_DOUBLE_EQ(final_speed, param_array[3]);
+}
+
+TEST(MovePrimTest, get_final_speed_test)
+{
+    const double final_speed = 2.11;
+
+    MovePrimitive move_prim = MovePrimitive(int(), Point(), Angle(), final_speed);
+
+    EXPECT_DOUBLE_EQ(move_prim.getFinalSpeed(), final_speed);
+}
+
+TEST(MovePrimTest, get_final_orientation_test)
+{
+    const Angle final_angle = Angle::ofRadians(3.15);
+
+    MovePrimitive move_prim = MovePrimitive(int(), Point(), final_angle, double());
+
+    EXPECT_EQ(move_prim.getFinalAngle(), final_angle);
+}
+
+TEST(MovePrimTest, get_final_destination_test)
+{
+    const Point destination = Point(-1, 2);
+
+    MovePrimitive move_prim = MovePrimitive(int(), destination, Angle(), double());
+
+    EXPECT_EQ(move_prim.getDestination(), destination);
+}
+
+TEST(MovePrimTest, get_extra_bit_array_test)
+{
+    MovePrimitive move_prim = MovePrimitive(int(), Point(), Angle(), double());
+
+    std::vector<bool> extra_bit_array = move_prim.getExtraBitArray();
+
+    EXPECT_EQ(extra_bit_array, std::vector<bool>());
+}
+
+TEST(MovePrimitiveTest, create_primitive_from_message_test)
+{
+    const Point destination  = Point(2, -3);
+    const Angle final_angle  = Angle::ofRadians(3.55);
+    const double final_speed = 2.22;
+    const int robot_id       = 3U;
```

Comment:
So you're assigning an unsigned int to an int. Consider declaring it unsigned int at the beginning.

Reply:
Set to unsigned.